main.js

```diff
@@ -3,10 +3,32 @@ var reservations = {
   'Ted': { claimed: true }
 }
 
-var name = prompt('Please enter the name for your reservation');
-
-var claimReservation = function () {
-  // write your code here!
+let namesArray=Object.keys(reservations)
+let c=0;
+var claimReservation= function(input){
+let name= document.getElementById("input").value;
+  for(let i of namesArray){
+  name = name.toLowerCase();
+  var x=i.toLowerCase();
+    if (name==x){
+      if(reservations[i].claimed==false){
+      document.getElementById("live").innerHTML = (i+" welcome");
+      reservations[i].claimed==true
+      }
+      else if(reservations[i].claimed==true){
+      document.getElementById("live").innerHTML = (i+" someone has alrady claimed your reservation");
+      }
+      c++;
+    }
+  }
+  if(c==0){
+  let newName = name
+  name = {claimed:true};
+  reservations[newName] = name
```

Comment:
instead of what you have on lines 25-27, you could just do this:
`let newUser = {name: name, claimed:true}`